### PR TITLE
tests: moving nested_exec to nested.tests exec

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change adds a new commands to nested.tests tool

The tests.nested now has 2 new commands:
 . exec
 . exec_as

### CODE REVIEW

File: tests/nested/core/core20-kernel-failover/task.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ execute: |
   echo "Wait for snapd to be available"
   retry=120
   wait=1
-  until nested_exec command -v snap; do
+  until tests.nested exec command -v snap; do
```
Comment (bboozzoo) [inline tests/nested/core/core20-kernel-failover/task.yaml:34]: probably a TODO, but this is part of the regular core vm bringup so we most likely don't nned it anymore. IIRC the test was written before we added those helpers.

File: tests/nested/core/core20-kernel-failover/task.yaml [hunk 2]
```diff
@@ -41,11 +41,11 @@ execute: |
   done
 
   echo "Wait for snapd to be seeded"
-  nested_exec sudo snap wait system seed.loaded
+  tests.nested exec sudo snap wait system seed.loaded
```
Comment (bboozzoo) [inline tests/nested/core/core20-kernel-failover/task.yaml:44]: same here, there's nested_exec "sudo snap wait seed.loaded" in nested.sh

**Review by bboozzoo — APPROVED** [review]
LGTM

**Comment by bboozzoo** [comment]
Does not seem to be related, but `google-nested:ubuntu-20.04-64:tests/nested/core/core20-gadget-reseal` prepare failed in a weird way:

```
Warning: Permanently added '[localhost]:8022' (ECDSA) to the list of known hosts.
Warning: Permanently added '[localhost]:8022' (ECDSA) to the list of known hosts.
Warning: Permanently added '[localhost]:8022' (ECDSA) to the list of known hosts.
Warning: Permanently added '[localhost]:8022' (ECDSA) to the list of known hosts.
/usr/bin/snap
Warning: Permanently added '[localhost]:8022' (ECDSA) to the list of known hosts.
sudo: snap: command not found
-----
```

we have nested_wait_for_snap_command which seems to have succeeded (hence `/usr/bin/snap` in the output), but on the call to `snap wait seed.loaded`

**Review by mardy — APPROVED** [review]
LGTM!

**Pull request merged**